### PR TITLE
DRILL-7812: updated hashCode function for MaterializedField

### PR DESCRIPTION
**Pull request by rymarm**
# [DRILL-7801](https://issues.apache.org/jira/browse/DRILL-7812): Broken equals/hashcode contract 

## Description
MaterializedField had broken hashCode/equals contract:

> If two objects are equal according to the equals(Object) method, then calling the hashCode method on each of the two objects must produce the same integer result.

In our case, equals are based on 2 parameters, while hashCode on 3. I updated hashCode and now it respond equals function.
`this.name.toLowerCase()` was added, because equals method compares name with ignoreCase.

## Testing
Manual tests

### CODE REVIEW

File: exec/vector/src/main/java/org/apache/drill/exec/record/MaterializedField.java [hunk 1]
```diff
@@ -189,7 +189,7 @@ public MaterializedField getOtherNullableVersion() {
 
   @Override
   public int hashCode() {
-    return Objects.hash(this.name, this.type, this.children);
+    return Objects.hash(this.name.toLowerCase(), this.type);
```
Comment (ihuzenko) [inline exec/vector/src/main/java/org/apache/drill/exec/record/MaterializedField.java:192]: is it guaranteed that name is never null?

**Review by ihuzenko — CHANGES_REQUESTED** [review]
Please add a unit test.

**Comment by cgivre** [comment]
@ihuzenko 
Could you please take another look and see if this is ready to be merged?
Thanks!

**Review by vdiravka — APPROVED** [review]
@rymarm Good catch. +1
Let's give @ihuzenko a few days to take a look before merge.

**Review by ihuzenko — APPROVED** [review]
Looks Great !!! Thanks  @rymarm for making the changes and sorry for my delay.

**Pull request merged**